catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +167,83 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService("dog_legs.sch", "dog_paws.sch");
+        try {
+            service.validate(getMetacard("dog_4leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            if (ex.getWarnings() != null) {
+                for (String warning : ex.getWarnings()) {
+                    assertThat(warning.equals(service.sanitize(warning)), is(true));
+                }
+            }
+            if (ex.getErrors() != null) {
+                for (String error : ex.getErrors()) {
+                    assertThat(error.equals(service.sanitize(error)), is(true));
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testSanitizationWithNoRemoveableDelimiters()
+            throws SchematronInitializationException {
+        SchematronValidationService service = getService(false,
+                "doggy-namespace",
+                true,
+                "dog_legs.sch",
+                "dog_paws.sch");
+
+        String s = "abcdefghijk";
+        assertThat(s, is(service.sanitize(s)));
+    }
+
+    @Test
+    public void testSanitizationWithNoRemoveableDelimitersAndTrimsALittleFromTheSides()
+            throws SchematronInitializationException {
+        SchematronValidationService service = getService(false,
+                "doggy-namespace",
+                true,
+                "dog_legs.sch",
+                "dog_paws.sch");
+
+        String s = " abcdefghijk ";
+        String t = "abcdefghijk";
+        assertThat(t, is(service.sanitize(s)));
+    }
+
+    @Test
+    public void testSanitizationWithRemoveableDelimiters()
+            throws SchematronInitializationException {
+        SchematronValidationService service = getService(false,
+                "doggy-namespace",
+                true,
+                "dog_legs.sch",
+                "dog_paws.sch");
+
+        String s = "abcd\tef ghij           k";
+        String t = service.sanitize(s);
+        assertThat(s, is(not(t)));
+        assertThat(service.sanitize(s), is(t));
```

Comment:
this assertion `assertThat(service.sanitize(s), is(t));` is an identity, `t` is literally `service.sanitize(s)` so it will always be true, even if it has incorrect behavior. In this case i think we want the output to be "abcd ef ghij k" but we don't verify that